auslib/test/test_db.py

```diff
@@ -1002,6 +995,132 @@ def testMergeUpdateWithConflict(self):
+                                         data_version=1)
+        self.db.permissions.t.insert().execute(permission="admin", username="bob", data_version=1)
+
+    def testAllTablesCreated(self):
+        self.assertTrue(self.table)
+        self.assertTrue(self.table.history)
+        self.assertTrue(self.table.scheduled_changes)
+        self.assertTrue(self.table.scheduled_changes.history)
+
+    def testSCTableHasCorrectColumns(self):
+        expected = set([
+            "sc_id", "scheduled_by", "complete", "when", "data_version", "base_fooid",
+            "base_foo", "base_bar", "base_data_version",
+        ])
+        columns = set([c.name for c in self.table.scheduled_changes.t.get_children()])
+        self.assertEquals(expected, columns)
+
+    def testSCTableWithNoConditions(self):
+        class TestTable2(AUSTable):
+
+            def __init__(self, db, metadata):
+                self.table = Table("test_table3", metadata, Column("fooid", Integer, primary_key=True, autoincrement=True),
+                                   Column("foo", String(15), nullable=False),
+                                   Column("bar", String(15)))
+                super(TestTable2, self).__init__(db, "sqlite", scheduled_changes=True, scheduled_changes_kwargs={"conditions": []},
+                                                 history=True, versioned=True)
+
+        self.assertRaisesRegexp(ValueError, "No conditions enabled", TestTable2, self.db, self.metadata)
+
+    def testSCTableWithBadConditions(self):
+        class TestTable3(AUSTable):
+
+            def __init__(self, db, metadata):
+                self.table = Table("test_table3", metadata, Column("fooid", Integer, primary_key=True, autoincrement=True),
+                                   Column("foo", String(15), nullable=False),
+                                   Column("bar", String(15)))
+                super(TestTable3, self).__init__(db, "sqlite", scheduled_changes=True, scheduled_changes_kwargs={"conditions": ["time", "blech"]},
+                                                 history=True, versioned=True)
+
+        self.assertRaisesRegexp(ValueError, "Unknown conditions", TestTable3, self.db, self.metadata)
+
+    def testValidateConditionsNone(self):
+        self.assertRaisesRegexp(ValueError, "No conditions found", self.sc_table._validateConditions, {})
+
+    @mock.patch("time.time", mock.MagicMock(return_value=200))
+    def testValidateConditionsJustWhen(self):
+        self.sc_table._validateConditions({"when": 12345678})
+
+    def testValidateConditionsTelemetryRaisesError(self):
+        conditions = {
+            "telemetry_product": "Firefox",
+            "telemetry_channel": "nightly",
+            "telemetry_uptake": "200000",
+        }
+        self.assertRaisesRegexp(ValueError, "uptake condition is disabled", self.sc_table._validateConditions, conditions)
```

Comment:
Not in the new ConfigurableConditions tests, but TestScheduledChangesTable has a couple: https://github.com/mozilla/balrog/blob/a300741332da939c1ba7eefa95e8c1cd147b1515/auslib/test/test_db.py#L701 -- does that cover it, or do you think we also need to test when _only_ telemetry is enabled?